libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -62,7 +61,13 @@ impl ConnectionManager {
 
     pub fn start(&mut self) -> Result<Connector, ConnectionManagerError> {
         let (sender, recv) = sync_channel(CHANNEL_CAPACITY);
-        let mut state = self.connection_state.clone();
+        let mut state = if let Some(state) = self.connection_state.take() {
```

Comment:
See my previous comment about this state.  The fact that `self.join_handle.is_some()` already tells us that the connection manager has already been started.